frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ImportCluster/ImportCluster.tsx

```diff
@@ -636,15 +710,49 @@ const AutoImportControls = (props: { state: State; dispatch: Dispatch<Action> })
         return t('import.mode.token')
       case ImportMode.kubeconfig:
         return t('import.mode.kubeconfig')
+      case ImportMode.discoveryROSA:
+        return t('import.mode.discovery')
     }
     return '-'
   }
 
+  const getImportModeHelperText = (m: ImportMode) => {
+    switch (m) {
+      case ImportMode.manual:
+        return t('import.mode.manual')
+      case ImportMode.discoveryROSA:
+        return t('import.mode.discovery.helpertext')
+      default:
+        return t('import.credential.explanation')
+    }
+  }
+
+  const secretIndex = resources.findIndex((item) => item.kind === 'Secret' && item?.metadata?.name === secretName)
+  const deleteCount = secretIndex >= 0 ? 1 : 0
+
+  const updateROSAImportSecret = useCallback(
+    (credentialName: string) => {
+      const newToken =
+        ocmCredentials.find((credential) => credential.metadata.name === credentialName)?.stringData?.ocmAPIToken ?? ''
+      const discoverySecret = cloneDeep(autoImportSecret)
+      discoverySecret.stringData = {
+        ...discoverySecret.stringData,
+        api_token: newToken,
+        cluster_id: clusterID,
+      }
+      discoverySecret.type = 'auto-import/rosa'
+      resources.splice(deleteCount ? secretIndex : 1, deleteCount, discoverySecret)
```

Comment:
Rather than repeating this logic here and moving the code that finds the index, could we have this function simply return the auto import secret, and do the splicing in the switch inside of `onChangeImportMode`? It just keeps the code that calculates the index and the code that uses it closer together, and it keeps the cases more parallel.